common-android/src/main/java/com/toggl/common/extensions/MapExtensions.kt

```diff
@@ -0,0 +1,7 @@
+package com.toggl.common.extensions
+
+fun <Key, Value> Map<Key, Value>.doesNotContainKey(key: Key) = containsKey(key).not()
+
+fun <Key, Value> Map<Key, Value>.addOrReplace(key: Key, value: Value): Map<Key, Value> =
+    if (doesNotContainKey(key)) this + (key to value)
+    else toMutableMap().also { it[key] = value }
```

Comment:
The stdlib's `+` operator already does what you want here.
If it's just for the readability, you can simplify this to:
```suggestion
    this + (key to value)
```

test:
```kotlin
        val map = mapOf(1 to "1", 2 to "2")
        map.addOrReplace(1, "0") shouldBe mapOf(1 to "0", 2 to "2")
        (map + (1 to "0")) shouldBe mapOf(1 to "0", 2 to "2")
```